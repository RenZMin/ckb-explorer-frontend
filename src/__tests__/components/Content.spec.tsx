import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Content from '../../components/Content'

describe('Content Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <Content children={<div className="content__child">content</div>} />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('.content__child')).toHaveLength(1)
    expect(wrapper.find('.content__child').text()).toBe('content')
  })
})
