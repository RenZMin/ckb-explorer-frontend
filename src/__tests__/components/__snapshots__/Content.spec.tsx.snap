// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component shallow renders 1`] = `
<div
  className="sc-fnVZcZ FgJVV"
>
  <div
    className="content__child"
  >
    content
  </div>
</div>
`;
